src/main/java/io/cdap/plugin/servicenow/restapi/RestAPIResponse.java

```diff
@@ -112,22 +128,32 @@ private static void validateRestApiResponse(String responseBody) {
     }
   }
 
-  private static void validateHttpResponse(HttpResponse response) {
+  private static ServiceNowAPIException validateHttpResponse(HttpResponse response) {
     int code = response.getStatusLine().getStatusCode();
     if (SUCCESS_CODES.contains(code)) {
-      return;
+      return null;
     }
     if (RETRYABLE_CODES.contains(code)) {
-      throw new RetryableException(String.format(HTTP_ERROR_MESSAGE, code));
+      return new ServiceNowAPIException(String.format(HTTP_ERROR_MESSAGE, code), response, true);
     }
-    throw new NonRetryableException(String.format(HTTP_ERROR_MESSAGE, code));
+    return new ServiceNowAPIException(String.format(HTTP_ERROR_MESSAGE, code), response, false);
```

Comment:
You can directly use
return new ServiceNowAPIException(String.format(HTTP_ERROR_MESSAGE, code), response, RETRYABLE_CODES.contains(code))